perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/GroupsManagerMethod.java

```diff
@@ -36,24 +44,60 @@
 	 * @param group Group JSON Group class
 	 * @return Group Newly created group
 	 */
+	/*#
+	 * Creates a new group in the specific VO defined by object vo in parameter.
+	 *
+	 * @param vo int Parent VO <code>id</code>
+	 * @param parentGroupId Integer parentGroupId
+	 * @param name String name of a group
+	 * @param description String description of a group
+	 * @return Group Newly created group
+	 */
 	createGroup {
 
 		@Override
 		public Group call(ApiCaller ac, Deserializer parms) throws PerunException {
 			ac.stateChangingCheck();
 
 			if (parms.contains("parentGroup")) {
-				return ac.getGroupsManager().createGroup(ac.getSession(),
-						ac.getGroupById(parms.readInt("parentGroup")),
-						parms.read("group", Group.class));
-			} else if (parms.contains("vo")) {
-				Group group = parms.read("group", Group.class);
-				if (group.getParentGroupId() == null) {
+				if (parms.contains("group")) {
 					return ac.getGroupsManager().createGroup(ac.getSession(),
-							ac.getVoById(parms.readInt("vo")),
+							ac.getGroupById(parms.readInt("parentGroup")),
+							parms.read("group", Group.class));
+				} else if (parms.contains("name") && parms.contains("description")) {
+					String name = parms.readString("name");
+					String description = parms.readString("description");
+					Group group = new Group(name, description);
+					return ac.getGroupsManager().createGroup(ac.getSession(),
+							ac.getGroupById(parms.readInt("parentGroup")),
 							group);
 				} else {
-					throw new RpcException(RpcException.Type.WRONG_PARAMETER, "Top-level groups can't have parentGroupId set!");
+					throw new RpcException(RpcException.Type.WRONG_PARAMETER);
+				}
+			} else if (parms.contains("vo")) {
+				if (parms.contains("group")) {
+					Group group = parms.read("group", Group.class);
+					if (group.getParentGroupId() == null) {
+						return ac.getGroupsManager().createGroup(ac.getSession(),
+								ac.getVoById(parms.readInt("vo")),
+								group);
+					} else {
+						throw new RpcException(RpcException.Type.WRONG_PARAMETER, "Top-level groups can't have parentGroupId set!");
+					}
+				} else if (parms.contains("parentGroupId") && parms.contains("name") && parms.contains("description")) {
```

Comment:
With ``parentGroupId`` present in params you will never get into this branch, since it's first condition on method start. I would rework this and retrieve/construct Group on method start based on ``group / name,description`` presence. Than you either have ``parentGroupId`` or ``vo`` to determine, if new group is a sub-group or top-level.